### PR TITLE
feat: [CDB-1265]: Telemetry Added to Dashboards

### PR DESCRIPTION
**Pull request by Harness-JKhattak**
### Summary

- Telemetry has been added to many key user events that happen throughout the Custom Dashboards app
- AIDA usage especially covered with successful and unsuccessful scenarios tracked

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by Harness-JKhattak** [comment]
trigger messagecheck

**Review by sean-harness — APPROVED** [review]

File: src/modules/45-dashboards/pages/home/DashboardForm.tsx [hunk 1]
```diff
@@ -77,6 +80,27 @@ const DashboardForm: React.FC<DashboardFormProps> = ({
...
+        action = CDBActions.DashboardCloneSubmitted
+        break
+    }
+
+    trackEvent(action, { category: Category.CUSTOM_DASHBOARDS })
+    onComplete({
+      ...completedFormData,
+      description: completedFormData.description?.join(TAGS_SEPARATOR) || ''
+    })
+  }
```
Comment (d1wilko) [inline src/modules/45-dashboards/pages/home/DashboardForm.tsx:102]: can I suggest passing `mode` as an arg to make this function pure?

**Review by d1wilko — APPROVED** [review]
1 suggestion, but I'll not let it block your PR

File: src/modules/45-dashboards/pages/home/GetStarted.tsx [hunk 2]
```diff
@@ -53,7 +51,7 @@ export const GetStarted: React.FC<DrawerProps> = props => {
           <Text font={{ size: 'medium', weight: 'semi-bold' }} color={Color.BLACK}>
             {getString('dashboards.getStarted.title')}
           </Text>
-          <Button icon="cross" minimal onClick={_ => props.setDrawerOpen(false)} />
+          <Button icon="cross" minimal onClick={_ => setDrawerOpen(false)} />
```
Comment (bgribben-harness) [inline src/modules/45-dashboards/pages/home/GetStarted.tsx:54]: Why the underscore?

File: src/modules/45-dashboards/pages/home/DashboardForm.tsx (hunk 1 above)
Comment (bgribben-harness) [reply src/modules/45-dashboards/pages/home/DashboardForm.tsx:102]: To make it pure you'd also need to pass all of `trackEvent` `onComplete` and `TAGS_SEPARATOR`

**Review by bgribben-harness — APPROVED** [review]

File: src/modules/45-dashboards/pages/home/DashboardForm.tsx (hunk 1 above)
Comment (Harness-JKhattak) [reply src/modules/45-dashboards/pages/home/DashboardForm.tsx:102]: If I do that then the `onSubmit={handleCompleteClicked}` formik function in the Button below becomes:

`onSubmit={(formData: DashboardFormikValues) => handleCompleteClicked(mode, formData)}`

Would the change be worth that?

File: src/modules/45-dashboards/pages/home/GetStarted.tsx (hunk 2 above)
Comment (Harness-JKhattak) [reply src/modules/45-dashboards/pages/home/GetStarted.tsx:54]: Good question, this component predates me on the team and hasn't really been touched.

Someone mustn't like writing `()`

**Pull request merged**